src/Heap.js

```diff
@@ -51,34 +77,61 @@ export default class Heap {
     }
 
     buildMinHeap() { 
-        const lastParent = Math.floor((this.heap.length - 2) / 2);
+        const lastParent = Math.floor((this.heap.length - 1) / 2);
 
         for (let i = lastParent; i >= 0; i--) {
             this.minHeapify(i);
         }
     }
 
     checkHeap(position) {
-        while (position> 0) {
-            position = Math.floor((position - 2) / 2);
+        while (position > 0) {
+            position = Math.floor((position - 1) / 2);
             this.minHeapify(position);
         }
     }
 
     insert(node, key) {
         this.heap.push({node, key});
         this._nodeIndices.set(node, this.heap.length - 1);
-        this.checkHeap(this.heap.length - 1);
+        this.filterUp(this.heap.length - 1);
     }
 
     compare(a, b) {
         if (a && b) {
-            return a.key - b.key;
+            if (Array.isArray(a) && Array.isArray(b)) {
+                if (a[0] > b[0] || (a[0] === b[0] && a[1] > b[1])) {
+                    return 1;
+                }
+
+                return -1;
+            }
+
+            return a - b;
         }
 
         return 0;
     }
 
+    remove(node) {
+        const index = this._nodeIndices.get(node);
+        if (index == null) {
+            return;
+        }
+        if (index === this.heap.length - 1) {
+            this._nodeIndices.delete(this.heap.pop().node);
+            return;
+        }
+
+        this.swap(index, this.heap.length - 1);
+        this._nodeIndices.delete(this.heap.pop().node);
+        if (index === 0 || this.compare(this.heap[Math.floor((index-1)/2)].key, this.heap[index].key) < 0) {
```

Comment:
hella whitespace, but also a bit of a verbose line
at the very least you could move computing the parent index to a different line
please don't forget that a larger amount of shorter lines is typically much better than one fat line
an 80 character width is typically a good measure for "this is starting to get pretty long" and i personally put a hard cap on 120